e2e/playwright/tests/todo_plugin.spec.ts

```diff
@@ -8,35 +8,71 @@
+  }
+  const dmURL = await getTodoBotDMPageURL(adminClient, '', adminUser.id);
+  await page.goto(dmURL, {waitUntil: 'load'});
+});
 
 export default {
-    connected: () => {
-        test.describe('available commands', () => {
-            test('with just the main command', async ({pages, page, pw}) => {
-               
-            const {adminClient, adminUser} = await pw.getAdminClient();
-            if (adminUser === null) {
-                throw new Error('can not get adminUser');
-            }
-            const dmURL = await getTodoBotDMPageURL(adminClient, '', adminUser.id);
-            await page.goto(dmURL, {waitUntil: 'load'});
-
-            const c = new pages.ChannelsPage(page);
-            const slash = new SlashCommandSuggestions(page.locator('#suggestionList'));
-
-            // # Run incomplete command to trigger help
-            await fillMessage('/todo', page);
-
-            // * Assert suggestions are visible
-            await expect(slash.container).toBeVisible();
-
-            // * Assert help is visible
-            await expect(slash.getItemTitleNth(0)).toHaveText('todo [command]');
-
-            await expect(slash.getItemDescNth(0)).toHaveText('Available commands: list, add, pop, send, settings, help');
-            });
-        });
-    },
+  setup: () => {
+    test('checking available commands', async ({ page }) => {
+      const slash = new SlashCommandSuggestions(page.locator('#suggestionList'));
+
+      // # Run command to trigger todo
+      await fillMessage('/todo', page);
+
+      // * Assert suggestions are visible
+      await expect(slash.container).toBeVisible();
+
+      // * Assert todo [command] is visible
+      await expect(slash.getItemTitleNth(0)).toHaveText('todo [command]');
+
+      await expect(slash.getItemDescNth(0)).toHaveText('Available commands: list, add, pop, send, settings, help');
+    });
+  },
+  commands: () => {
+    test('help', async ({ pages, page, pw }) => {
+      const c = new pages.ChannelsPage(page);
+
+      // # Run command to trigger help
+      postMessage('/todo help', page);
+
+      // # Grab the last post
+      const post = await getLastPost(page);
+      const postBody = post.locator('.post-message__text-container');
+
+      // * Assert /todo add [message] command is visible
+      await expect(postBody).toContainText('add [message]');
+
+      // * Assert /todo list command is visible
+      await expect(postBody).toContainText('list');
+
+      // * Assert /todo list [listName] command is visible
+      await expect(postBody).toContainText('list [listName]');
+
+      // * Assert /todo pop command is visible
+      await expect(postBody).toContainText('pop');
+
+      // * Assert /todo send [user] [message] command is visible
+      await expect(postBody).toContainText('send [user] [message]');
+
+      // * Assert /todo settings summary [on, off] command is visible
+      await expect(postBody).toContainText('settings summary [on, off]');
```

Comment:
@hanzei Curious what you think about this test. It could be a bit brittle since it's checking the output in this much detail. Essentially if _anything_ changes about the help text (other than the order of commands in the output), this test will break.

We would also want to make this test fail if a new command is added to the help text, so maybe we want to count the number of lines somehow?

I think this is a difficult problem to solve, since we mainly want e2e tests to be "this feature generally works correctly", rather than "this feature's output looks exactly like this"